### DISCUSSION TITLE
How to use pod_override and call other class?

### DISCUSSION THREAD

**Discussion by sunny760408** [apache/airflow#22268] [Q&A] [answered]
Hi everyone,

I have a DAG as below, a function "task_with_template" call a class "CustomerOperator".
When not use Decorator (@task(executor_config=executor_config_template)) DAG is executable.
But when  use Decorator, call  class "CustomerOperator" is not work.
I would like to know how to modify code that can use pod_override and call other class?

```
from apple.get.operators import CustomerOperator
import os
from airflow.settings import AIRFLOW_HOME
from airflow.decorators import task

from airflow import DAG
from airflow.operators.python import PythonOperator
from airflow.utils.dates import days_ago
from airflow.providers.amazon.aws.hooks.s3 import S3Hook
from datetime import datetime, timedelta
from kubernetes.client import models as k8s

default_args = {
    'owner':'apple',
    'email_on_failure':False,
    'email':['apple@test.com'],
}

with DAG(
    dag_id='apple_dag', 
    start_date = days_ago(1), 
    schedule_interval=None,
    catchup=False,
    default_args=default_args,
    access_control={
        'apple':{'can_read', 'can_edit'}        
    },
    tags=['apple']

) as dag:
    executor_config_template = {
        "pod_template_file": os.path.join(AIRFLOW_HOME, "pod_templates/apple_pod.yaml"),
        "pod_override": k8s.V1Pod(metadata=k8s.V1ObjectMeta(labels={"release": "stable"})),
    }

    @task(executor_config=executor_config_template)
    def task_with_template():
        getData= CustomerOperator(
            task_id='apple_prd', 
            type='prd'
        )

    test = task_with_template()
```

**Comment by potiuk** [comment, marked as answer]
No idea what the "CustomerOperator" is, but decorators and "Operators" do not work together. They are simply different way of doing the same thing (making custom code execute as Airflow tasks. In case of decorated tasks, the custom code is inside the decorated function. In case of CustomOperators it is in the operator..

Instead of reusing operators you can extract common code in form of functions from your common libraries, or you can use Hooks inside decorated tasks instead of using Custom Operators (which is probably what you want). In this case the code you share is in Hook rather than operator (and it rather should be this way.

**Comment by sunny760408** [comment]
Hi Potiuk,

Thanks for your reply. 
I'm not pretty understand OOPs concept, you really help me a lot.
